### DISCUSSION TITLE
Fail2ban regex identifies bad IP addresses but doesn't ban them in one jail, but does in others

### DISCUSSION THREAD

**Discussion by spassat** [fail2ban/fail2ban#3888] [Q&A]
I run an Ubuntu server 24.04 LTS with Fail2ban v 1.1.0.1 installed.

I had difficulty with an earlier version of Fail2ban after upgrading to 24.04 LTS, which were resolved (see https://www.airix.net/index.php?option=com_content&view=article&id=89:ubuntu-2404-with-fail2ban-issue-fix&catid=53:featured&Itemid=271).

I now have a strange situation whereby Fail2ban correctly identifies and bans bad IP addresses via the postfix jail...BUT, although it identifies bad IP addresses via the dovecot jail, it does not ban them.

Thus, running this command:
```

fail2ban-regex /var/log/mail.log /etc/fail2ban/filter.d/dovecot.conf[mode=aggressive]

```
...results in over 3,000 matches, but none of the related IP addresses are banned. By comparison, matches of the mail.log with the postfix.conf filter identifies and then bans the respective matches. 

The entries in `/etc/fail2ban/jail.local` are:
```
[dovecot]

enabled  = true
mode     = aggressive
filter   = dovecot
logpath  = /var/log/mail.log
protocol = all
maxretry = 2

[postfix]
enabled  = true
mode     = aggressive
maxretry = 2
protocol = all
logpath  = /var/log/mail.log
```
Any ideas?

**Comment by sebres** [comment]
See the FAQ in https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works

To get banned, the IP needs to be found with `maxretry` attempts in `findtime` seconds.
For the other issues take a look in above mentioned FAQ.

**Comment by spassat** [comment]
Thanks, sebres. My settings with regard to ```mexretry``` and ```findtime``` are fine. However, I did find the answer to the question: the problem results from a banned IP address matching two regex conditions for two jails.

In that situation, it only bans once, making it appear that the second match is not working. In fact, the offending IP address is banned across all ports, so in practice the ban prevents the offender connecting to the server.

**Reply by sebres** [reply]
No. Finding the failures or banning in jail1 doesn't affect jail2 and vice versa... Thus ban across two jails is completely independent from each other and must happen. 
Take a look at FAQ I provided above... Maybe it is something else, e. g. like wrong backend (jail monitoring systemd journal instead of logfile), etc...
